src/libsyntax/util/small_vector.rs

```diff
@@ -69,6 +69,42 @@ impl<T> SmallVector<T> {
         }
     }
 
+    pub fn pop(&mut self) -> Option<T> {
+        match self.repr {
+            Zero => None,
+            One(..) => {
+                let one = mem::replace(&mut self.repr, Zero);
+                match one {
+                    One(v1) => Some(v1),
+                    _ => unreachable!()
+                }
+            }
+            Many(..) => {
+                let mut many = mem::replace(&mut self.repr, Zero);
+                let item =
+                    match many {
```

Comment:
I wasn't sure whether it was a concern about allocation time or space (I guessed space, from the name).  I also went for the safe thing, since I wasn't sure if the rest of the impl was prepared for zero- or one-length vectors, rather than Zero and One enum values.

It would definitely simplify things, if that is an acceptable way to do it.